05-onchain.md

```diff
@@ -325,252 +328,265 @@ transaction output.
 to spend the HTLC output (using a payment preimage) is for it to use the
 HTLC-success transaction.
 
-The payment preimage either serves to prove payment (when the offering node
-originated the payment) or to redeem the corresponding incoming HTLC from
-another peer (when the offering node is forwarding the payment). Once a node has
-extracted the payment, it no longer cares about the fate of the HTLC-spending
-transaction itself.
+The payment preimage either serves to prove payment (when the offering node is
+the originator of the payment) or to redeem the corresponding incoming HTLC from
+another peer (when the offering node is forwarding the payment). After a node has
+extracted the payment, it no longer need be concerned with the fate of the
+HTLC-spending transaction itself.
 
 In cases where both resolutions are possible (e.g. when a node receives payment
-success after timeout), either interpretation is acceptable; it is the
+success after timeout), either interpretation is acceptable: it's the
 responsibility of the recipient to spend it before this occurs.
 
-Once it has timed out, the HTLC output needs to be spent (to prevent
-the remote node using the HTLC-success transaction) before the local node can
+Once it has timed out, the local node needs to spend the HTLC output (to prevent
+the remote node from using the HTLC-success transaction) before it can
 back-fail any corresponding incoming HTLC, using `update_fail_htlc`
 (presumably with reason `permanent_channel_failure`), as detailed in
 [BOLT 02](https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#forwarding-htlcs).
 If the incoming HTLC is also on-chain, a node simply waits for it to
-timeout: there's no way to signal early failure.
+timeout, as there's no way to signal early failure.
 
-If an HTLC is too small to appear in *any* commitment transaction, it
+If an HTLC is too small to appear in *any commitment transaction*, it
 can be safely failed immediately. Otherwise,
-if a HTLC isn't in the *local* commitment transaction a node needs to make sure
+if a HTLC isn't in the *local commitment transaction* a node needs to make sure
 that a blockchain reorganization or race does not switch to a
-commitment transaction which does contain it before the node fails it, hence
+commitment transaction that does contain it before the node fails it: hence
 the wait. The requirement that the incoming HTLC be failed before its
 own timeout still applies as an upper bound.
 
 ## HTLC Output Handling: Remote Commitment, Remote Offers
 
-Each HTLC output can only be spent by us, the recipient, using the payment
-preimage. If the local node does not possess the preimage (and doesn't discover
-it), it's the offerer's responsibility to spend the HTLC output, once it's timed
+Each HTLC output can only be spent by the recipient by using the payment
+preimage. If a node does not possess the preimage (and doesn't discover
```

Comment:
The double-by is just a slight bit awkward. Maybe "If it uses the payment preimage" instead?

Reply:
changed